### PR TITLE
Add docs to clarify project root heuristics

### LINKED ISSUE #4674: `isort` forgetting 1st party below sub-config
In today's version of monorepo hurdles, here is a minimal repro repo with `ruff=0.0.267` and `numpy==1.24.3` installed:

```none
.
├── folder_1
│   └── module.py
├── folder_2
│   ├── main.py
│   ├── module.py
│   └── ruff.toml
└── pyproject.toml
```

```toml
# pyroject.toml

[tool.ruff]

select = ["ALL"]
ignore = [
    "D2",
    "INP001",
]
```

```toml
# ruff.toml
```

```python
"""folder_1/module.py."""

SOME_OTHER_MODULE_IMPORT = 0
```

```python
"""folder_2/module.py."""

SOME_SAME_MODULE_IMPORT = 0
```

```python
"""folder_2/main.py."""

import time  # noqa: F401

import numpy as np  # noqa: F401

from folder_1.module import SOME_OTHER_MODULE_IMPORT  # noqa: F401
from folder_2.module import SOME_SAME_MODULE_IMPORT  # noqa: F401
```

All is well, `ruff` reports no issues from the repo root:

```none
repo > ruff .
```

### Issue

Now, changing `ruff.toml` to be:

```toml
# ruff.toml

extend = "../pyproject.toml"
```

Suddenly we get:

```none
repo > ruff .
folder_2/main.py:3:1: I001 [*] Import block is un-sorted or un-formatted
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

So, what's the issue about?
- With sub-config `ruff.toml` without`extend`: `ruff` from root correctly infers `folder_1` as 1st party
- With subconfig `ruff.toml` using `extend`: `ruff` from root now infers `folder_1` as 3rd party

I think sub-config extensions should inherit 1st party understandings from parent config.  What do you think?

### PR DESCRIPTION
**Pull request by charliermarsh**
Closes #4674.

### CODE REVIEW

**Comment by charliermarsh** [comment]
@jamesbraza - Open to feedback!

File: docs/faq.md [hunk 1]
```diff
@@ -255,6 +255,52 @@ src = ["src", "tests"]
...
 
+## How does Ruff determine which of my imports are first-party, third-party, etc.?
+
+Ruff accepts a `src` option that in your `pyproject.toml`, `ruff.toml`, or `.ruff.toml` file, which
+specifies the directories that Ruff should consider when determining whether an import is
+first-party.
+
+For example, if you have a project with the following structure:
+
+```
```
Comment (calumy) [inline docs/faq.md:266]:
```suggestion
```tree
```

Hopefully this will fix the ci issue

**Comment by charliermarsh** [comment]
Merging but can always follow-up on any feedback.

**Review by jamesbraza — COMMENTED** [review]
Looks great, thank you!

One trick I personally found useful was using `--verbose --no-cache`, so that `ruff` prints out why it makes 1st vs 3rd party decisions.  If you want, that can be added to the new FAQ section.

Otherwise, thanks again!

**Pull request merged**